### DISCUSSION TITLE
Provide a GUI.

### DISCUSSION THREAD

**Discussion by beedell-roke** [espanso/espanso#1755] [Feature requests and Ideas]
I can't subscribe solely to issue closure in a discussion, and considering that per [this Reddit post](https://www.reddit.com/r/espanso/comments/gomu73/a_sneak_peek_of_the_upcoming_gui_for_those_who/?utm_source=share&utm_medium=web2x&context=3#:~:text=freddytstudio-,A%20Sneak%20Peek%20of%20the%20upcoming%20GUI%20for%20those%20who%20are%20interested%20:\),-r/espanso%20%2D%20a), it *is* being worked upon, any discussion should be about implementation detais - what an *issue* excels at.

**Comment by AucaCoyan** [comment]
Hi! Nice to see you here. I would like to ask you if you have some specific need of the GUI, or anything that could be of significance to the front end?
Otherwise, I would close the issue if it were for me.

**Comment by beedell-roke** [comment]
> Hi! Nice to see you here.

Thanks, @AucaCoyan. Likewise.

> I would like to ask you if you have some specific need of the GUI?

My reason for wanting a GUI is that all the people who would benefit from this the most, the bureaucrats (teachers and police officers) I've interacted with, including my family, would be perfectly competent enough and willing enough to use Espanso to create shortcuts, since they really need a solution to repetitive data input, but won't touch a commandline, even a TUI. Most of the time they aren't even allowed to at work.

> anything that could be of significance to the front end?

I'd like to know what it's being written in, since for me (wx)Qt is the best because I use KDE Plasma and Windows, neither of which GTk (for instance) is any good on.

**Comment by AucaCoyan** [comment]
Perfect! That gives more info to the devs, thanks!
As Federico said [here](https://github.com/espanso/espanso/issues/1276), feature requests are moved to discussions so they can be voted and prioritized properly 😄 
See you there!

**Reply by beedell-roke** [reply]
> feature requests are moved to discussions so they can be voted and prioritized properly

Can you elaborate, @AucaCoyan? How do GitHub Discussions benefit this better than Issues do?

**Reply by AucaCoyan** [reply]
It's a decision made by Federico, the owner and almost entire creator of the app, very few people committed by. I'm still new to the app, so even my opinion is not completely well informed.
Going to the point, every repo manages issues differently, and this is the way we chose to manage feature requests. It "loses" notification customization, but "wins" votes from the users, and we value more the community feedback than if we close or not a _feature request_ issue (not bugs!)
Aside from that, there is already another discussion about the GUI, but I decided to keep both anyways 😄

**Reply by beedell-roke** [reply]
> https://github.com/espanso/espanso/discussions/1755#discussioncomment-7669223

@AucaCoyan, can't votes also be provided on issues? That's the bit I didn't quite understand in his post too.

**Reply by AucaCoyan** [reply]
Nope, GitHub issues can't be voted. You can checkout the docs about [issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and [discussions](https://docs.github.com/en/discussions)

**Reply by beedell-roke** [reply]
> https://github.com/espanso/espanso/discussions/1755#discussioncomment-7669271

@AucaCoyan, the issue you linked (https://github.com/espanso/espanso/issues/1276#issue-1292045417) has been voted upon by 3 users. It is a feature I use every day. Consequently, we must have different definitions of voting, although I'm unable to imagine what yours is.

**Reply by AucaCoyan** [reply]
I'm sorry RokeJulian, you are _a_ user as every other person using this app. Everybody has a first priority for anything.

And I reiterate: issues can't be voted. A thumb up or down is not a vote.

**Reply by beedell-roke** [reply]
> And I reiterate: issues can't be voted. A thumb up or down is not a vote.

Apologies; I wasn't aware you'd stated this. I wouldn't have asked a (supposedly) second time if I knew. What does count as a vote, then, if not the thumbs up and down? In every other repository, especially https://github.com/microsoft/vscode/issues?q=is%3Aissue. See https://github.com/microsoft/vscode/issues/57687#issuecomment-574911131:

> This feature request is now a candidate for our backlog. The community has 60 days to upvote the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it.

The upvote emoji reaction is used to indicate support there. What is used in this repository?

---

> I'm sorry RokeJulian, you are a user as every other person using this app. Everybody has a first priority for anything.

@AucaCoyan, I don't know what you're saying here.

**Reply by AucaCoyan** [reply]
That's it. I cannot afford more time explaining things more clearly. Espans is a open source project and it will remain so. People spend their free time trying to improve it.
If you want, you can message federico to ask for a contribution role, and stop asking the same questions over and over.
I ask you to _read_ the pages and you still come up with different arguments for your special case that is more important that the whole community.
Please try to be kind to the users that do this for free

**Reply by beedell-roke** [reply]
> https://github.com/espanso/espanso/discussions/1755#discussioncomment-7673086

I have no idea of what you're referring to, @AucaCoyan. Obviously it's FOSS - it's under the https://github.com/espanso/espanso#GPL-3.0-1-ov-file (per https://github.com/espanso/espanso/blob/72b516c0ef53ab10dc5289da3355f300af875cda/LICENSE#L1). I don't believe we've ever mentioned that, much less either of us disputed it. Why are you talking about that instead of just answering what you mean by "a vote"? You're confusing me - apologies.

> If you want, you can message federico to ask for a contribution role, and stop asking the same questions over and over.

I don't want that role. I've never contributed. What questions do I keep supposedly asking? I don't understand what you're being so affrontive about - I'm just asking about the voting system. This is silly.

If, for instance, you don't actually know how the voting system works, just tell me that, and I'll try to work it out myself or ask someone who does. This is weird.

**Comment by federico-terzi** [comment]
@RokeJulianLockhart Thanks for the feedback! Have you checked out this GUI made from one of our community members: https://github.com/unobserved-io/espanso-gui

An official GUI is not currently being worked on at the moment (due to lack of available time), but for sure the project would benefit from it :)

**Reply by beedell-roke** [reply]
> https://github.com/espanso/espanso/discussions/1755#discussioncomment-7673111

No, but that's brilliant, @federico-terzi. Thanks lots for the suggestion, and for your work on the project. I'm surprised I missed that - I can now suggest the tool!

**Reply by beedell-roke** [reply]
> https://github.com/espanso/espanso/discussions/1755#discussioncomment-7673111

@federico-terzi, since you're a maintainer, do you suggest that I open an issue about this now that https://github.com/espanso/espanso/issues/1276#issuecomment-2430346481 has been reverted, or is that project best kept separate?
